controllers/nexus/resource/deployment/manager.go

```diff
@@ -28,12 +28,13 @@ import (
 
 	"github.com/m88i/nexus-operator/api/v1alpha1"
 	"github.com/m88i/nexus-operator/pkg/framework"
+	"github.com/m88i/nexus-operator/pkg/framework/kind"
 	"github.com/m88i/nexus-operator/pkg/logger"
 )
 
 var managedObjectsRef = map[string]resource.KubernetesResource{
-	framework.DeploymentKind: &appsv1.Deployment{},
-	framework.ServiceKind:    &corev1.Service{},
+	kind.DeploymentKind: &appsv1.Deployment{},
+	kind.ServiceKind:    &corev1.Service{},
```

Comment:
I would avoid creating too many packages. What's the problem with `framework` for common k8s API features?

Reply:
Needed that to avoid an import cycle: the fake client now needs the Kinds in order to populate the API Resources list from each GroupVersion, but the tests from `framework` actually use the fake client. So these had to be moved to a separate package.

But why do you want to avoid having too many packages?

Reply:
> But why do you want to avoid having too many packages?

[Keep this in mind](https://dave.cheney.net/practical-go/presentations/gophercon-singapore-2019.html#_consider_fewer_larger_packages).

Reply:
That's reasonable. What if we merge the `test` package into `framework`?

Reply:
the `test` package shouldn't be included in the final binary, so it's reasonable to keep it there. That's ok having `kind`, but I'll keep an eye on new packages creation, just to make sure we are not abusing.